ci/ci/ci.py

```diff
@@ -214,13 +214,29 @@ def refresh_deploy_jobs(jobs):
                 try_to_cancel_job(job)
     prs.refresh_from_deploy_jobs(latest_jobs)
 
+
+@app.route('/force_retest_flat', methods=['POST'])
+def force_retest_flat():
+    d = request.json
+    source = FQRef(Repo(d['source_repo_owner'], d['source_repo_name']), d['source_branch_name'])
+    target = FQRef(Repo(d['target_repo_owner'], d['target_repo_name']), d['target_branch_name'])
+    return _force_retest(source, target)
```

Comment:
This will redirect the user to an empty page after pressing the button. I think this endpoint should redirect to `redirect('/ui')` instead of returning `'', 200`.

Reply:
https://i.pinimg.com/originals/62/18/bb/6218bb5d10fccd219a1b4119eb6b92e2.jpg